docs/advanced_examples/KNearestNeighbors.ipynb

```diff
@@ -0,0 +1,296 @@
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import time\n",
+    "\n",
+    "import pandas as pd\n",
+    "from sklearn.datasets import make_classification\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "from tqdm.auto import tqdm\n",
+    "\n",
+    "from concrete.ml.sklearn import KNeighborsClassifier as ConcreteKNeighborsClassifier"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Data-set generation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "X, y = make_classification(\n",
+    "    n_samples=13, n_features=20, n_informative=3, n_redundant=0, n_classes=2, n_clusters_per_class=1\n",
+    ")\n",
+    "# Split the data-set into a train and testing sets\n",
+    "X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Model instantiation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# The novel aspect introduced by Concret-ML models is the hyperparameter `n_bits`, which represents\n",
+    "# the precision for quantizing input data and intermediate values of the model.\n",
+    "# This quantization step is essential after the training phase, since FHE exclusively operates\n",
+    "# over integers.\n",
+    "concrete_knn = ConcreteKNeighborsClassifier(n_bits=3, n_neighbors=3)\n",
+    "\n",
+    "# Fit both the Concrete ML and its equivalent float estimators on clear data\n",
+    "concrete_knn, sklearn_model = concrete_knn.fit_benchmark(X_train, y_train)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Compile the model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Saved in 66\n",
+      "Compilation time: 2.60 seconds\n"
+     ]
+    }
+   ],
+   "source": [
+    "# The compilation step aims to:\n",
+    "# - convert the quantized model to its FHE equivalent\n",
+    "# - create an executable operation graph\n",
+    "# - check the operation graph's compatibility with FHE\n",
+    "# - compute the maximum bit-width needed for model execution\n",
+    "# - determine cryptographic parameters necessary for generating secret keys and evaluation keys\n",
+    "time_begin = time.time()\n",
+    "circuit = concrete_knn.compile(X)\n",
+    "print(f\"Compilation time: {time.time() - time_begin:.2f} seconds\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Generating a key for an 9-bit circuit\n"
```

Comment:
Should we use round PBS? I think it's a great feature here as it's pretty hard to control the accumulator.

Reply:
Let's just make sure there is no crash and keep accumulator at 6-7 bits.

Reply:
I'll add back the round PBS here.

Reply:
We now need to discuss the design part, in which part rounding_threshold_bits would be most appropriate, either in :
- the init
- the compile 
- the fit